robocorp-code/vscode-client/src/inspector.ts

```diff
@@ -0,0 +1,198 @@
+    let inspectorLaunchInfo = await commands.executeCommand(ROBOCORP_GET_LANGUAGE_SERVER_PYTHON_INFO) as InterpreterInfo | undefined;
+    if (!inspectorLaunchInfo) {
+            OUTPUT_CHANNEL.appendLine("Unable to get Robocorp Inspector launch info.");
+            return;
+        }
+
+    // add locators.json path to args
+    if(locatorJson) args.push("--database", locatorJson);
+
+    // if locatorType is given prioritize that. Else Ensure that a locator is selected!
+    if (locatorType) {
+        args.push("add");
+        args.push(locatorType)
+    } else {
+        const locatorSelected: LocatorEntry | undefined = locator || getSelectedLocator(
+          "Please select a locator first.",
+          "Please select only one locator.",
+        );
+        if (locatorSelected) args.push("edit", locatorSelected.name)
+        else {
+            OUTPUT_CHANNEL.appendLine("Unable to open Robocorp Inspector. Select a locator first.");
+            return;
+        }
+    }
+    window.withProgress({
+          location: ProgressLocation.Notification,
+          title: "Robocorp",
+          cancellable: false,
+      },
+      (progress) => {
+        progress.report({ message: 'Opening Inspector...' });
+        return new Promise<void>(resolve => {
+              setTimeout(() => {
+                  resolve();
+              }, 3000);
+          })
+      });
+    OUTPUT_CHANNEL.appendLine(`Args for inspector: ${args.join(", ")}`);
+    const launchResult: ExecFileReturn = await startInspectorCLI(
+      inspectorLaunchInfo.pythonExe,
+      args,
+      inspectorLaunchInfo.environ);
+    OUTPUT_CHANNEL.appendLine('Inspector CLI stdout:');
+    OUTPUT_CHANNEL.appendLine(launchResult.stdout);
+    OUTPUT_CHANNEL.appendLine('Inspector CLI stderr:');
+    OUTPUT_CHANNEL.appendLine(launchResult.stderr);
+}
+
+
+
+export async function getInspectorPythonInfo(): Promise<InterpreterInfo | undefined> {
+    if (_cachedInspectorPythonInfo) {
+        OUTPUT_CHANNEL.appendLine('Using CACHED getInspectorPythonInfo');
+        return _cachedInspectorPythonInfo;
+    }
+    OUTPUT_CHANNEL.appendLine('Fetch getInspectorPythonInfo');
+    let cachedInspectorPythonInfo = await getInspectorPythonInfoUncached();
+    if (!cachedInspectorPythonInfo) {
+        return undefined; // Unable to get it.
+    }
+    // Ok, we got it (cache that info).
+    _cachedInspectorPythonInfo = cachedInspectorPythonInfo;
+    return _cachedInspectorPythonInfo;
+}
+
+
+async function startInspectorCLI(pythonExecutable: string, args: string[], environ?: { [key: string]: string }): Promise<ExecFileReturn> {
+    const inspectorCmd = ['-m', 'inspector.cli'];
+    const completeArgs = inspectorCmd.concat(args)
+    return execFilePromise(
+      pythonExecutable, completeArgs,
+      {env: {...process.env, ...environ}}
+    );
+}
+
+const getInspectorPythonInfoUncached = async (): Promise<InterpreterInfo | undefined> => {
```

Comment:
This shouldn't be needed (I guess this is some leftover... you're already using `ROBOCORP_GET_LANGUAGE_SERVER_PYTHON_INFO` in this module, so the `ROBOCORP_GET_INSPECTOR_PYTHON_INFO` can probably be completely removed (I can't see any usage of it).

Reply:
ah, yes. Forgot to cleanup the "old" inspector environment management out of the way. Good catch!

Reply:
Removed